src/validators/patch.ts

```diff
@@ -0,0 +1,51 @@
+import { APIGatewayProxyEvent } from 'aws-lambda';
+import { addHttpHeaders } from '../util/httpHeaders';
+
+export function validateUpdateVinRequest(event: APIGatewayProxyEvent) {
+  if (!event.body) {
+    return addHttpHeaders({
+      statusCode: 400,
+      body: JSON.stringify({ error: 'invalid request' }),
+    });
+  }
+
+  if (!event.headers.Authorization) {
+    return addHttpHeaders({
+      statusCode: 400,
+      body: JSON.stringify({ error: 'Missing authorization header' }),
+    });
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  const newVin: string = JSON.parse(event.body).newVin as string;
+
+  if (!newVin) {
+    return addHttpHeaders({
+      statusCode: 400,
+      body: JSON.stringify({ error: 'You must provide a new VIN' }),
+    });
+  }
+  return undefined;
+}
+
+export function validateVins(oldVin: string, newVin: string) {
+  if (
+    !newVin
+    || newVin.length < 3
+    || newVin.length > 21
+    || typeof newVin !== 'string'
+  ) {
+    return addHttpHeaders({
+      statusCode: 400,
+      body: JSON.stringify({ error: 'New VIN is invalid' }),
+    });
+  }
+  if (newVin === oldVin) {
```

Comment:
Genuine question, do we care if a user wants to do this? Seems like if you want to be stupid, you can?

Reply:
A good question, I mean it would just be a bit of a waste of time, you'd be archiving the current and creating a new record for no real reason? I have no stronger reason than it just feels like a waste of time